src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/CourtService.kt

```diff
@@ -147,4 +148,44 @@ class CourtService(
       hearingType = type,
       madeByTheCourt = specification.madeByTheCourt
     )
+
+    @Transactional
+    fun deleteVideoLinkBooking(videoBookingId: Long) {
+      val booking = videoLinkBookingRepository.findById(videoBookingId).orElseThrow {
+        EntityNotFoundException("Video link booking with id $videoBookingId not found")
+      }
+
+      booking.toAppointments().forEach { prisonApiService.deleteAppointment(it.appointmentId) }
+      booking.toAppointments().forEach { videoLinkAppointmentRepository.deleteById(it.id!!) }
```

Comment:
You don't need to delete the appointments.  JPA cascades should delete the appointments when you delete the booking.

Reply:
reved delete operations